core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java

```diff
@@ -320,6 +325,53 @@ public void truncateTable(String namespace, String table) throws ExecutionExcept
     }
   }
 
+  @Override
+  public void createIndex(
+      String namespace, String table, String columnName, Map<String, String> options)
+      throws ExecutionException {
+    TableMetadata tableMetadata = getTableMetadata(namespace, table);
+    TableMetadata newTableMetadata =
+        TableMetadata.newBuilder(tableMetadata).addSecondaryIndex(columnName).build();
+
+    updateIndexingPolicy(namespace, table, newTableMetadata);
+
+    // update metadata
+    putTableMetadata(namespace, table, newTableMetadata);
+  }
+
+  @Override
+  public void dropIndex(String namespace, String table, String columnName)
+      throws ExecutionException {
+    TableMetadata tableMetadata = getTableMetadata(namespace, table);
+    TableMetadata newTableMetadata =
+        TableMetadata.newBuilder(tableMetadata).removeSecondaryIndex(columnName).build();
+
+    updateIndexingPolicy(namespace, table, newTableMetadata);
+
+    // update metadata
+    putTableMetadata(namespace, table, newTableMetadata);
+  }
+
+  private void updateIndexingPolicy(
+      String databaseName, String containerName, TableMetadata newTableMetadata)
+      throws ExecutionException {
+    CosmosDatabase database = client.getDatabase(databaseName);
+    try {
+      // get the existing container properties
+      CosmosContainerResponse response =
+          database.createContainerIfNotExists(containerName, PARTITION_KEY_PATH);
+      CosmosContainerProperties properties = response.getProperties();
+
+      // set the new index policy to the container properties
+      properties.setIndexingPolicy(computeIndexingPolicy(newTableMetadata));
+
+      // update the container properties
+      database.getContainer(containerName).replace(properties);
+    } catch (RuntimeException e) {
+      throw new ExecutionException("dropping the secondary index failed", e);
```

Comment:
Maybe, not necessarily for dropping?

Reply:
Ah, it looks like the error message is wrong. I will fix it. Thank you!